build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```diff
@@ -182,6 +182,12 @@ fun Test.jvmVersionForTest(): JavaLanguageVersion {
     return JavaLanguageVersion.of(getPropertyFromAnySource("testJavaVersion").getOrElse(JavaVersion.current().majorVersion))
 }
 
+fun Test.configureTestQuarantine() {
+    if (project.providers.gradleProperty("flakyTestQuarantine").isPresent) {
+        jvmArgumentProviders.add(CommandLineArgumentProvider { listOf("-Dorg.gradle.flakyTestQuarantine=true") })
```

Comment:
Why use a command line argument provider here? The flag is an actual input to the test task, so we should treat it like that.